### PR TITLE
replace time.clock with perf_counter for py3

### PR DESCRIPTION
**Pull request by frostming**
Replace the timer to suppress the deprecation warning.

### CODE REVIEW

File: flask_sqlalchemy/__init__.py [hunk 1]
```diff
@@ -36,7 +36,10 @@
 
 # the best timer function for the platform
 if sys.platform == 'win32':
-    _timer = time.clock
+    if sys.version_info()[:2] > (3, 3):
```
Comment (chardbury) [inline flask_sqlalchemy/__init__.py]: See: https://docs.python.org/3/library/time.html#time.perf_counter

```suggestion
    if sys.version_info()[:2] >= (3, 3):
```

Comment (frostming) [reply flask_sqlalchemy/__init__.py]: Oh, thanks for getting into this.

**Comment by MichaelF91** [comment]
I came here to suggest this, as I'm getting ugly deprecation warnings using pytest. Is there a reason why this pull request hasn't yet been accepted?

**Comment by rsyring** [comment]
#662 uses `if sys.version_info >= (3, 3):` while this PR uses `sys.version_info[:2] >= (3, 3):`.  I believe both would work given tuple comparison semantics and the former is just a bit cleaner.

Also, we don't currently have CI setup and I don't have easy access to a Windows machine.  Will someone manually confirm that this code works as expected on Windows for Python 2.7 and a version of Python3 > 3.3.

**Comment by frostming** [comment]
@rsyring I squashed the commits and tested on Windows PC, with Python 3.7.2 and Python 2.7.15, all pass. Thanks

**Comment by rsyring** [comment]
Great, thank you.  I'll update the milestone to 2.4 as I see no reason to delay this.

**Comment by rsyring** [comment]
Would you mind rebasing this on the current `2.x-maintenance` branch and change the PR to merge into that branch?  I believe that is the last step needed before I can get this merged.

Thanks.

**Comment by davidism** [comment]
The command for that would be `git rebase --onto origin/2.x-maintenance origin/master deprecate-timer`, then `git push -f`, then edit the PR here to change the target.

**Comment by rsyring** [comment]
@davidism ok, I'll take care of it then.  Thanks @frostming for the submission.

**Comment by rsyring** [comment]
@frostming I've replaced this PR with #700 b/c I wasn't sure if I could rebase and force-push to your fork's branch.

**Comment by rsyring** [comment]
@davidism I merged this PR to the maintenance branch, after rebasing again, but the PR didn't pick up that the commit was merged.  Let me know here or on Discord if you can tell what I did wrong.

Closing this PR, but it was merged.

**Pull request closed without merging**